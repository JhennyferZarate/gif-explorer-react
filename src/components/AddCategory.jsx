import { useState } from "react";
import PropTypes from "prop-types";

const formatInput = (value) => {
  const words = value.trim().split(" ");
  const formattedWords = words.map(
    (word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
  );
  return formattedWords.join(" ");
};

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    onNewCategory(formatInput(inputValue));
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="form" className="input-box">
      <i className="fa fa-search" aria-hidden="true"></i>
      <input
        type="text"
        placeholder="Search gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
