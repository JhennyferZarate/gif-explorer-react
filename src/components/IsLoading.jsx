export const IsLoading = () => {
  return (
    <div className="loading-section">
      <div className="loading-spinner"></div>
      <p>Loading...</p>
    </div>
  );
};
