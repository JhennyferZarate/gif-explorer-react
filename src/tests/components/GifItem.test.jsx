import { render, screen } from "@testing-library/react";
import { GifItem } from "../../components/GifItem";

describe("GifItem Tests", () => {
  const title = "Image Title GIF";
  const url = "https://img.com/imagen.jpg";

  test("Should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("Should show the image with url and ALT indicated", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");

    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Should show the title in the component", () => {
    render(<GifItem title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
