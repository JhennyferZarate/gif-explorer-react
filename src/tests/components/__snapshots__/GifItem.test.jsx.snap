// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Tests Should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Image Title GIF"
      src="https://img.com/imagen.jpg"
    />
    <p>
      Image Title GIF
    </p>
  </div>
</div>
`;
